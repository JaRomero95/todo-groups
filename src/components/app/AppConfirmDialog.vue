<template>
  <v-dialog
    :value="true"
    persistent
  >
    <v-card>
      <v-card-title>
        {{ title }}
      </v-card-title>
      <v-card-text>
        {{ message }}
      </v-card-text>
      <v-card-actions>
          <v-spacer></v-spacer>

          <v-btn
            flat="flat"
            @click="$emit('cancel')"
            data-test="dialog-cancel-btn"
          >
            Cancel
          </v-btn>

          <v-btn
            color="success"
            flat="flat"
            @click="$emit('confirm')"
            data-test="dialog-confirm-btn"
          >
            Confirm
          </v-btn>
        </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: true
    },
    message: {
      type: String,
      required: true
    }
  }
}
</script>
