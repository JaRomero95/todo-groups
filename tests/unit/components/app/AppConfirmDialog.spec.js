import {shallowMount} from '@vue/test-utils';
import AppConfirmDialog from '@/components/app/AppConfirmDialog.vue';

const title = 'Hello world';
const message = 'Lorem ipsum dolor';

const options = {
  propsData: {
    title,
    message,
  }
};

describe('AppConfirmDialog', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AppConfirmDialog, options);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show title prop', () => {
    const text = wrapper.find('v-card-title-stub').text();
    expect(text).toEqual(title);
  });

  it('show message prop', () => {
    const text = wrapper.find('v-card-text-stub').text();
    expect(text).toEqual(message);
  });

  it('cancel button emit cancel event', () => {
    wrapper.find('[data-test="dialog-cancel-btn"]').vm.$emit('click');
    expect(wrapper.emitted().cancel).toBeTruthy();
  });

  it('confirm button emit confirm event', () => {
    wrapper.find('[data-test="dialog-confirm-btn"]').vm.$emit('click');
    expect(wrapper.emitted().confirm).toBeTruthy();
  });
});
