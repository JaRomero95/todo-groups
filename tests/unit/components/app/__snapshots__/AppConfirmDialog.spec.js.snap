// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppConfirmDialog renders correctly 1`] = `
<v-dialog-stub contentclass="" value="true" persistent="true" maxwidth="none" origin="center center" width="auto" transition="dialog-transition">
  <v-card-stub tag="div">
    <v-card-title-stub>
      Hello world
    </v-card-title-stub>
    <v-card-text-stub>
      Lorem ipsum dolor
    </v-card-text-stub>
    <v-card-actions-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub activeclass="v-btn--active" tag="button" flat="true" type="button" data-test="dialog-cancel-btn">
        Cancel
      </v-btn-stub>
      <v-btn-stub color="success" activeclass="v-btn--active" tag="button" flat="true" type="button" data-test="dialog-confirm-btn">
        Confirm
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
